File: C#/JsonParser.cs
using UserSpace; // Assuming this is the correct namespace for your User class
using System.Text.Json;

namespace JsonParser
{
    class JsonUserParser
    {
        public static List<User> ReadUsersFromJson(string filename)
        {
            List<User> users = new List<User>();
            try
            {
                // Read the JSON file into a string
                string jsonString = File.ReadAllText(filename);

                var options = new JsonSerializerOptions
                {
                    PropertyNameCaseInsensitive = true
                };

                // Deserialize the JSON string into a List of User objects
                Root root = JsonSerializer.Deserialize<Root>(jsonString, options) ?? new Root();
                if (root.users != null) // Ensure 'users' is not null before accessing it
                {
                    users = root.users;
                }
            }
            catch (Exception e)
            {
                Console.WriteLine("Exception: " + e.Message);
            }
            return users; // Return the populated list of users
        }
    }
}

File: python/JSONparser.py
import json
from user import User

def read_users_from_json(filename):
    with open(filename, 'r') as file:
        data = json.load(file)  # Load the JSON data into a Python dictionary
        users = []
        for item in data['users']:  # Iterate through the list under the 'users' key
            # Create a User instance for each user in the list
            users.append(User(int(item['id']), item['name'], item['email']))
    return users

# Assuming the file name is "file4.json" and is in the correct path
users = read_users_from_json("../files/file4.json")

for user in users:
    print(user.printUser())
